app/views/PickerView.tsx

```diff
@@ -59,14 +61,19 @@ const Item = React.memo(({ item, selected, onItemPress, theme }: IItem) => (
 	/>
 ));
 
-const RenderSearch = ({ hasSearch, onChangeText }: IRenderSearch) => {
+const RenderSearch = ({ hasSearch, onChangeText, searchText, onCancelSearch }: IRenderSearch) => {
 	if (!hasSearch) {
 		return <List.Separator style={styles.listNoHeader} />;
 	}
 	return (
 		<>
 			<View style={styles.search}>
-				<SearchBox onChangeText={onChangeText} />
+				<SearchBox
+					showCancelIcon={searchText.length > 0}
+					onCancelSearch={onCancelSearch}
+					onChangeText={onChangeText}
+					value={searchText}
+				/>
```

Comment:
You shouldn't have to change other places like this.
Searchbox should be able to handle its own state.